components/src/core/hero/hero.component.ts

```diff
@@ -23,24 +26,24 @@ import { Component, Input, OnInit } from '@angular/core';
         </div>
     `,
     styleUrls: ['./hero.component.scss'],
-    inputs: ['color', 'label', 'value', 'units', 'fontSize', 'iconSize'],
 })
-export class HeroComponent implements OnInit {
+export class HeroComponent implements OnChanges {
     @Input() color: string;
     @Input() label: string;
     @Input() value: string;
     @Input() units: string;
-    @Input() iconSize = 'normal';
-    @Input() iSize = 36;
-    @Input() fontSize = 'normal';
+    @Input() iconSize: IconSize = 'normal';
+    @Input() fontSize: FontSize = 'normal';
+    @Input() iconBackgroundColor: string;
+    iSize = 36;
 
-    ngOnInit(): void {
-        // We can't support dynamic iconSize w/ px-blue icons until https://github.com/angular/components/issues/5188 is fixed
+    // We can't support dynamic iconSize w/ px-blue icons until https://github.com/angular/components/issues/5188 is fixed
+    ngOnChanges(): void {
         this.iSize =
             this.iconSize === 'large'
                 ? 72
                 : this.iconSize === 'normal'
                 ? 36
-                : (this.iSize = Math.max(10, Math.min(72, parseInt(this.iconSize, 10))));
+                : this.iconSize;
```

Comment:
should we allow this at all if it doesn't work with PX Blue icons? Or just stick to the two default options?

Reply:
I wanted to match the flexibility of the React HeroComponent; it allows an icon of any size.  I'm personally not a fan of just allowing for two sizes; it's good to have them as options but not as a limitation.  I made a defect to address this limitation.

Reply:
There is an issue for this already on the icons repo (https://github.com/pxblue/icons/issues/27). I'm not sure we should support this until it actually works...if a user tries to change the size they're going to see this:
```
<pxb-hero [label]="'Temperature'" [value]="'98'" [units]="'°F'" [iconSize]="100">
    <mat-icon svgIcon="px-icons:temp" primary></mat-icon>
</pxb-hero>
<pxb-hero [label]="'Humidity'" [value]="'54'" [units]="'%'" [iconSize]="100">
    <mat-icon mat-list-icon primary>cloud</mat-icon>
</pxb-hero>
```
![image](https://user-images.githubusercontent.com/29152776/78247025-1f878100-74b8-11ea-9323-9c38ca7c1ad5.png)

Reply:
I've reverted the hero back to accepting two icon sizes.  I want to revisit this in the future so that either more sizes are supported or custom sizing is possible.